Objects/frameobject.c

```diff
@@ -246,7 +246,13 @@ mark_stacks(PyCodeObject *code_obj, int len)
                     next_stack = pop_value(pop_value(pop_value(next_stack)));
                     stacks[i+1] = next_stack;
                     break;
-
+                case SEND:
+                    j = get_arg(code, i) + i + 1;
+                    assert(j < len);
+                    assert(stacks[j] == UNINITIALIZED || stacks[j] == pop_value(next_stack));
```

Comment:
Is this correct?  assertions will be removed in debug builds and `pop_value()` has side-effects.

Reply:
No side effects on `int`s.

Reply:
(Disregard the comment. I should have looked at what `pop_value()` does.)

Reply:
Well, `pop_value` name implies that it modifies a stack (I suspect that's what @lpereira means as a side effect). In reality though it's rather `get_stack_top`:

https://github.com/python/cpython/blob/d4363d214097b3fca8b7910c2e0e91c8b0873fb2/Objects/frameobject.c#L167-L171

https://github.com/python/cpython/blob/cb8f491f46e262549f6c447b31625cab7c20a60a/Include/pyport.h#L296-L301